tests/test_filters.py

```
@@ -0,0 +1,234 @@
+        assert "connection_string_with_credentials" in hits
+
+    def test_env_assignment(self):
+        hits = detect_secrets("DB_PASS=supersecret123")
+        assert "env_secret_assignment" in hits
+
+    def test_no_secrets_in_clean_content(self):
+        hits = detect_secrets("User prefers concise responses in English.")
+        assert hits == []
+
+    def test_never_echoes_raw_secret(self):
+        raw_secret = "sk-abc123def456ghi789jkl012mno345pqr678"
```

Ruff (0.15.20)

[error] Possible hardcoded password assigned to: "raw_secret"

(S105)